projects/packages/external-connections/src/class-external-connections.php

```diff
@@ -0,0 +1,305 @@
+	 */
+	public static function get_connection( $service ) {
+		if ( ( new Host() )->is_wpcom_simple() ) {
+			require_lib( 'external-media-service/external-media-list' );
+			require_lib( 'external-connections' );
+
+			$connections = \WPCOM_External_Connections::init();
+			$token       = \ExternalMediaService::get_service_token( $service, get_current_user_id() );
+
+			if ( ! empty( $token->unique_id ) ) {
+				return $connections->get_keyring_connection_item( $token->unique_id );
+			}
+		} else {
+			$response = Client::wpcom_json_api_request_as_user( '/me/connections' );
+			if ( is_wp_error( $response ) ) {
+				return null;
+			}
+			$body = json_decode( wp_remote_retrieve_body( $response ) );
+			if ( isset( $body->connections ) && is_array( $body->connections ) ) {
+				foreach ( $body->connections as $connection ) {
+					if ( $service === $connection->service ) {
+						return (array) $connection;
+					}
+				}
+			}
+		}
+
+		return null;
+	}
+
+	/**
+	 * Checks if a connection to the specified service exists and is active.
+	 *
+	 * @param string $service The service identifier.
+	 *
+	 * @return bool True if an active connection exists, false otherwise.
+	 */
+	public static function has_connection( $service ) {
+		$connection = self::get_connection( $service );
+
+		if ( empty( $connection ) ) {
+			return false;
+		}
+
+		$connection_status = $connection['status'] ?? 'not_connected';
+		return $connection_status === 'ok';
+	}
+
+	/**
+	 * Deletes a connection for the provided service.
+	 *
+	 * @param string $service The service identifier.
+	 */
+	public static function delete_connection( $service ) {
+		$connection = self::get_connection( $service );
+		if ( empty( $connection ) ) {
+			return;
+		}
+
+		if ( ( new Host() )->is_wpcom_simple() ) {
+			if ( get_current_user_id() === $connection['user_ID'] ) {
+				require_lib( 'external-connections' );
+				$connections = \WPCOM_External_Connections::init();
+				$connections->delete_keyring_connection( $connection['ID'] );
+			}
+		} else {
+			Client::wpcom_json_api_request_as_user(
+				'/me/connections/' . $connection['ID'],
+				'2',
+				array( 'method' => 'DELETE' )
+			);
+		}
+	}
+
+	/**
+	 * Registers connection settings.
+	 */
+	public static function register_settings() {
+		foreach ( self::$services as $page => $services ) {
+			global $pagenow;
+
+			if ( $pagenow !== "options-$page.php" ) {
+				continue;
+			}
+
+			$host = new Host();
+
+			if ( ! $host->is_wpcom_simple() ) {
+				$connection = new Connection_Manager( 'jetpack' );
+				$status     = new Status();
+
+				if ( $status->is_offline_mode() ) {
+					return;
+				}
+
+				if ( ! $connection->has_connected_owner() ) {
+					return;
+				}
+
+				if ( ! $connection->is_user_connected() ) {
+					return;
+				}
+			}
+
+			add_settings_section(
+				'external_connections_section',
+				__( 'Integrations', 'jetpack-external-connections' ),
```

Comment:
Should this have "Jetpack" prefix?

Reply:
06ebc4532a251154871203f16ae95075c66b9b3c